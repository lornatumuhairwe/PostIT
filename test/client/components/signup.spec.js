import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Signup from '../../../client/app/components/authentication/signup';

test('it renders the signup page correctly', () => {
  const handleUsernameChange = jest.fn();
  const handleEmailChange = jest.fn();
  const handlePasswordChange = jest.fn();
  const handleConfirmPasswordChange = jest.fn();
  const handleSignUpAction = jest.fn();

  const rendered = renderer.create(
    <MemoryRouter initialEntries={['/signup']}>
      <Signup
        handleUsernameChange={handleUsernameChange}
        handleEmailChange={handleEmailChange}
        handlePasswordChange={handlePasswordChange}
        handleConfirmPasswordChange={handleConfirmPasswordChange}
        handleSignUpAction={handleSignUpAction}
      />
  </MemoryRouter>);
  expect(rendered.toJSON()).toMatchSnapshot();
});
