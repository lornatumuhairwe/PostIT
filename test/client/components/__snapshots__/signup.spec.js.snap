// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the signup page correctly 1`] = `
<div>
  <div
    className="ui inverted vertical center aligned segment"
  >
    <div
      className="ui container"
    >
      <div
        className="ui large inverted pointing secondary menu"
      >
        <a
          className="item"
          href="/"
          onClick={[Function]}
        >
          PostIt
        </a>
        <div
          className="right item"
          onClick={[Function]}
        >
          <a
            className="ui inverted button"
            disabled={undefined}
            href="/login"
            onClick={[Function]}
            role="button"
            style={
              Object {
                "marginLeft": "0.5em",
              }
            }
            tabIndex={undefined}
          >
            Login
          </a>
          <a
            className="ui active inverted button"
            disabled={undefined}
            href="/signup"
            onClick={[Function]}
            role="button"
            style={
              Object {
                "marginLeft": "0.5em",
              }
            }
            tabIndex={undefined}
          >
            Sign up
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="login-form"
  >
    <div
      className="ui center aligned middle aligned grid"
      style={
        Object {
          "height": "100%",
        }
      }
    >
      <div
        className="column"
        style={
          Object {
            "marginTop": 100,
            "maxWidth": 450,
          }
        }
      >
        <h2
          className="ui teal center aligned header"
        >
          Signup for an account
        </h2>
        <form
          action={undefined}
          className="ui large form"
          onSubmit={[Function]}
        >
          <div
            className="ui stacked segment"
          >
            <div
              className="field"
            >
              <div
                className="ui fluid left icon input"
                content={undefined}
              >
                <input
                  disabled={undefined}
                  onChange={[Function]}
                  placeholder="Username"
                  required={undefined}
                  tabIndex={undefined}
                  type="text"
                />
                <i
                  aria-hidden="true"
                  className="user icon"
                />
              </div>
            </div>
            <div
              className="field"
            >
              <div
                className="ui fluid left icon input"
                content={undefined}
              >
                <input
                  disabled={undefined}
                  onChange={[Function]}
                  placeholder="Email"
                  required={undefined}
                  tabIndex={undefined}
                  type="email"
                />
                <i
                  aria-hidden="true"
                  className="lock icon"
                />
              </div>
            </div>
            <div
              className="field"
            >
              <div
                className="ui fluid left icon input"
                content={undefined}
              >
                <input
                  disabled={undefined}
                  onChange={[Function]}
                  placeholder="Password"
                  required={undefined}
                  tabIndex={undefined}
                  type="password"
                />
                <i
                  aria-hidden="true"
                  className="lock icon"
                />
              </div>
            </div>
            <div
              className="field"
            >
              <div
                className="ui fluid left icon input"
                content={undefined}
              >
                <input
                  disabled={undefined}
                  onChange={[Function]}
                  placeholder="Confirm Password"
                  required={undefined}
                  tabIndex={undefined}
                  type="password"
                />
                <i
                  aria-hidden="true"
                  className="lock icon"
                />
              </div>
            </div>
            <button
              className="ui teal large fluid button"
              disabled={undefined}
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
            >
              Signup
            </button>
          </div>
        </form>
        <div
          className="ui message"
        >
          Already have an account? 
          <a
            href="/login"
          >
            Sign In
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
